tests/src/python/test_qgsgeometry.py

```diff
@@ -4459,6 +4459,36 @@ def testDeleteVertexCurvePolygon(self):
         expected_wkt = "CurvePolygon (CompoundCurve (CircularString (0 0, 1 1, 2 0),(2 0, 0 0)))"
         self.assertEqual(geom.asWkt(), QgsGeometry.fromWkt(expected_wkt).asWkt())
 
+    def testConvertVertex(self):
+
+        # WKT BEFORE -> WKT AFTER A CONVERT ON POINT AT 10,10
+        test_setup = {
+            # Curve
+            'LINESTRING(0 0, 10 0, 10 10, 0 10, 0 0)': 'COMPOUNDCURVE((0 0, 10 0), CIRCULARSTRING(10 0, 10 10, 0 10), (0 10, 0 0))',
+            'COMPOUNDCURVE(CIRCULARSTRING(0 0, 10 0, 10 10, 0 10, 0 0))': 'COMPOUNDCURVE(CIRCULARSTRING(0 0, 10 0, 10 10, 0 10, 0 0))',
+            'COMPOUNDCURVE((0 0, 10 0), CIRCULARSTRING(10 0, 10 10, 0 10), (0 10, 0 0))': 'COMPOUNDCURVE((0 0, 10 0, 10 10, 0 10, 0 0))',
```

Comment:
This means that toggling the same vertex twice *does not* bring back the original geometry! The linestring has been replaced by a compoundcurve!

Reply:
Yes indeed, but these are equivalent (e.g. `QgsGeometry.fromWkt("LINESTRING(0 0, 10 10, 20 0)").isGeosEqual(QgsGeometry.fromWkt("COMPOUNDCURVE((0 0, 10 10, 20 0))")) # True`)